src/mozblocklist.js

```diff
@@ -285,6 +285,16 @@ export default class Mozblocklist {
    * @param {boolean} options.selfsign          If true, signing will occur using the shared key.
    */
   async signBlocklist({ pending=null, selfsign=false }) {
+    console.warn("Signing blocklist...");
+    if (selfsign) {
+      if (await this.kinto.getBlocklistStatus() === "work-in-progress") {
+        await this.kinto.reviewBlocklist();
+      }
+      await this.kintoapprover.signBlocklist();
+    } else {
+      await this.kinto.signBlocklist();
+    }
+
```

Comment:
Moving this block up makes sure we stage first (if there is anything to stage), and afterwards fetch the preview.

Reply:
This doesn't cover the case where pending is passed in, which happens for reviewAndSignBlocklist. I also wanted to move the question to remove the security group up front so you don't have to wait so long to answer the question.

Reply:
Also, wouldn't this make `res = []` when the block is self-signed and `pending = null` ?

Reply:
Should be fixed in d611cb7ff9724d98b4114f65813fe236b9b12309.